pkg/services/ngalert/api/api_testing_test.go

```diff
@@ -0,0 +1,284 @@
+package api
+
+import (
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	models2 "github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/accesscontrol"
+	acMock "github.com/grafana/grafana/pkg/services/accesscontrol/mock"
+	"github.com/grafana/grafana/pkg/services/datasources"
+	"github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/services/ngalert/eval"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/web"
+)
+
+func TestRouteTestGrafanaRuleConfig(t *testing.T) {
+	t.Run("when fine-grained access is enabled", func(t *testing.T) {
+		rc := &models2.ReqContext{
+			Context: &web.Context{
+				Req: &http.Request{},
+			},
+			SignedInUser: &models2.SignedInUser{
+				OrgId: 1,
+			},
+		}
+
+		t.Run("should return 401 if user cannot query a data source", func(t *testing.T) {
+			data1 := models.GenerateAlertQuery()
+			data2 := models.GenerateAlertQuery()
```

Comment:
Can these be elevated to the shared setup code like `rc`? (applies throughout)

Reply:
It can but I am not sure it will give us any benefit other than save a few line of code.